ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -1255,10 +1252,7 @@ private void applyTtlUpdate(MessageInfo messageInfo, RemoteReplicaInfo remoteRep
   private void applyUndelete(MessageInfo messageInfo, RemoteReplicaInfo remoteReplicaInfo) throws StoreException {
     DataNodeId remoteNode = remoteReplicaInfo.getReplicaId().getDataNodeId();
     try {
-      messageInfo = new MessageInfo(messageInfo.getStoreKey(), messageInfo.getSize(), messageInfo.isDeleted(),
-          messageInfo.isTtlUpdated(), true, messageInfo.getExpirationTimeInMs(), messageInfo.getCrc(),
-          messageInfo.getAccountId(), messageInfo.getContainerId(), messageInfo.getOperationTimeMs(),
-          messageInfo.getLifeVersion());
+      messageInfo = new MessageInfo.Builder(messageInfo).isUndeleted(true).isDeleted(false).build();
```

Comment:
looks like this one has slightly different behavior. The builder code always sets `isDeleted` to false. Was this intended?

Reply:
Good catch, I made a mistake, isDeleted shouldn't be changed, will update.